src/main/scala/org/ergoplatform/explorer/persistence/dao/TransactionDao.scala

```diff
@@ -0,0 +1,39 @@
+package org.ergoplatform.explorer.persistence.dao
+
+import doobie.free.connection.ConnectionIO
+import doobie.implicits._
+import doobie.refined.implicits._
+import doobie.util.fragment
+import org.ergoplatform.explorer.{Id, TxId}
+import org.ergoplatform.explorer.persistence.models.Transaction
+import fs2.Stream
+
+object TransactionDao extends Dao {
+
+  val tableName: String = "node_transactions"
+
+  val fields: List[String] = List(
+    "id",
+    "header_id",
+    "coinbase",
+    "timestamp",
+    "size"
+  )
+
+  def getMain(id: TxId): ConnectionIO[Option[Transaction]] =
+    (selectWithRefFr("t") ++ leftJoinHeadersOnIdFr ++ fr"where h.main_chain = true")
```

Comment:
In this code you don't actually abstract from DB schema. 
Let's have plain and clear SQL statement here (parameterised if necessary), for which we will know for sure it is sent to the database.